Design rules/Waarden.md

```diff
@@ -0,0 +1,142 @@
+
+_**Ratio:**_
+Een developer moet bij het coderen begrijpen wat de code betekent, om fouten in de implementatie te voorkomen.
+
+## DR2.8 Overweeg een enumeratiewaarde weer te geven als string in een bevraging-API
+
+Het toepassen van een enumratie brengt het risico van tight coupling en breaking changes met zich mee.
+
+Indien een attribuut in het informatiemodel voorkomt als enumeratie, maar de corresponderende property komt alleen voor in de response op een bevraging om aan gebruikers te worden getoond dan is het niet nodig het ook in de API specificaties als een enumeratie te definiëren.
+
+Wanneer de enumeratiewaarde gebruikt wordt in code (algoritmes), definieer dan betekenisvolle maar bondige enumeratiewaarden.
+
+_**Ratio:**_
+
+Het opnemen van een enumeratie in de API-definitie betekent dat voor het kunnen tonen van een nieuwe of gewijzigde waarde in de enumeratie een nieuwe versie van de API moet worden uitgebracht. Voor bevraging-API's is dat een ongewenste vorm van tight coupling.
+
+Als het gegeven echter wordt gebruikt in code (algoritme) dan is het van belang dat er niet afgeweken kan worden van de bekende waarden en is een definitie als enumeratie op zijn plaats.
+
+## DR2.9 Neem een indicator op om de betekenis "magic strings" expliciet te maken.
+
+Wanneer het gevuld zijn van een datum functionele betekenis heeft, ook wanneer deze volledig onbekend is, wordt een indicator opgenomen om dit aan te geven. Als bv een datum veld de waarde "00000000" bevat en dat betekenis heeft wordt niet deze ongeldige waarde opgenomen in een date-veld, maar wordt een indicator toegevoegd.
+
+Indien in het onderstaande voorbeeld er helemaal niets bekend is van de overlijdensdatum, maar er is wel bekend dat de persoon overleden is, dan kan dat nergens uit worden afgeleid. Uit de reponse kan je niet opmaken of de overlijdensdatum leeg is omdat iemand  niet is overleden of omdat de overlijdensdatum niet bekend is.
+
+```
+    Persoon:
+      type: object
+      description: "Natuurlijkpersoon of NietNatuurlijkPersoon"
+      properties:
+        identificatie:
+          type: string
+        datumOverleden:  
+          $ref: "#/components/schemas/TypePersoonEnum"type: string
+        voorletters:
+          type: string
+```
+
+Door voor een dergelijke situatie een boolean op te nemen waar de betreffende informatie wordt gerepresenteerd wordt duidelijk wat de situatie is zonder dat er invalide waardes hoeven te worden opgenomen in de datum-velden.
+
+```
+    Persoon:
+      type: object
+      description: "Natuurlijkpersoon of NietNatuurlijkPersoon"
+      properties:
+        identificatie:
+          type: string
+        datumOverleden:  
+          $ref: "#/components/schemas/TypePersoonEnum"type: string
+        indicatieOverleden:
+          type: boolean
+        voorletters:
+          type: string
+```
+
+_**Ratio:**_
```

Comment:
Hier geef je als Ratio een voorbeeld, de ratio is volgens mij echter nooit een voorbeeld maar legt uit waarom we dit zo doen. Hier zou je i.i.g. al aan kunnen geven dat we op deze wijze voldoen aan DR4.2 maar wellicht kunnen we codeertechnische redenen aanvoeren waarom dit beter is. Zo vermoed ik dat tegen deze wijze van modelleren eenvoudiger te programmeren is.